app/models/bot/alegre.rb

```diff
@@ -505,46 +506,91 @@ def self.return_prioritized_matches(pm_id_scores)
-      # 2. A is confirmed to B and C is confirmed to B: type of the relationship between A and C is: confirmed
-      # 3. A is suggested to B and C is suggested to B: type of the relationship between A and C is: suggested
-      # 4. A is suggested to B and C is confirmed to B: type of the relationship between A and C is: suggested
+      # 1. A is confirmed to B and C is suggested to B: type of the relationship between A and C is: suggested to A
+      # 2. A is confirmed to B and C is confirmed to B: type of the relationship between A and C is: confirmed to A
+      # 3. A is suggested to B and C is suggested to B: type of the relationship between A and C is: IGNORE (don't suggest to suggest)
+      # 4. A is suggested to B and C is confirmed to B: type of the relationship between A and C is: form new relationship to B, break old relation to A
       parent_relationship = parent_relationships.first
+      proposed_relationship_is_confirmed = pm_id_scores[proposed_id][:relationship_type] == Relationship.confirmed_type
       new_type = Relationship.suggested_type
-      if parent_relationship.is_confirmed? && pm_id_scores[parent_id][:relationship_type] == Relationship.confirmed_type
+
+      # (3) if relationship to parent was suggested, and new relationship is also suggested
+      # we don't want to record this any more https://meedan.atlassian.net/browse/CV2-2675
+      if !parent_relationship.is_confirmed? & !proposed_relationship_is_confirmed
+        Rails.logger.info "[Alegre Bot] [ProjectMedia ##{pm.id}] [Relationships 3/6] [Relationships WARNING] ignoring suggested relationship pm_id, pm_id_scores, parent_id #{
+          [pm.id, pm_id_scores, proposed_id].inspect}"
+        return
+      end
+
+      # (1,2) relationships look great, but the proposed match should be replaced by a match to its parent
+      if parent_relationship.is_confirmed? 
+        if proposed_relationship_is_confirmed
+          new_type = Relationship.confirmed_type
+        end
+        original_parent_id = proposed_id
+        parent_id = parent_relationship.source_id
+        original_relationship = pm_id_scores[original_parent_id]
+        pm_id_scores[parent_id] = pm_id_scores[original_parent_id]
+        pm_id_scores[parent_id][:relationship_type] = new_type if pm_id_scores[parent_id]
+      end
+
+      # (4) if the relationship to parent was only suggested, but new relationship is confirmed
+      if !parent_relationship.is_confirmed? & proposed_relationship_is_confirmed
         new_type = Relationship.confirmed_type
+        # break the old parent relationship involving proposed_id, make the proposed_id into a new parent
+        Rails.logger.info "[Alegre Bot] [ProjectMedia ##{pm.id}] [Relationships 3/6] [Relationships NOTE] removing suggested relationship pm_id, parent_id #{
+          [parent_relationship.source_id, parent_relationship.target_id].inspect}"
+        parent_relationship.destroy
```

Comment:
I suggest to use `destroy!` instead of `destroy` so we can get an Errbit exception if something goes wrong.

Reply:
for my education, what does the `!` do here?  ruby docs suggest it is kind of like a warning that this is modifying the state of the object?